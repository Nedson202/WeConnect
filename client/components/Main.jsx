import React from 'react';
import { Link } from 'react-router-dom';

/**
   * @description Create main content in homepage.
   * 
   * @returns {object} JSX object
   * 
   * @memberof Main
   */
const Main = () => {
  return (
    <div>
      <div className="text-center" id="main-content">
        <h1>Welcome to WeConnect</h1>
        <h3>Here, we make it easier for you to reach businesses faster</h3>
        <Link to="/businesses" className="btn btn-outline-success">Explore</Link>
      </div>
    </div>
  );
}

export default Main;
