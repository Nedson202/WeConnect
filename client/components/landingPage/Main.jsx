import React from 'react';
import { Link } from 'react-router-dom';

/**
   * @description Create main content in homepage.
   *
   * @returns {object} JSX object
   *
   * @memberof Main
   */
const Main = () => (
  <div>
    <div className="text-center" id="main-content">
      <h1>Welcome to WeConnect</h1>
      <h3>Here, we make it easier for you to reach businesses faster</h3>
      <Link to="/businesses"><button className="btn" id="explore-button">Explore</button></Link>
    </div>
  </div>
);

export default Main;
