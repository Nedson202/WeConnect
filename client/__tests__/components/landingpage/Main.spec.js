import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import Main from '../../../components/landingPage/Main';

configure({ adapter: new Adapter() });

describe('Component: App', () => {
  it('should render', () => {
  const wrapper = shallow(<Main />);
    expect(wrapper.find('div').length).toBe(2);
    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper.find('h1').text()).toBe('Welcome to WeConnect');
    expect(wrapper.find('h3').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('button').text()).toBe('Explore');
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
    expect(wrapper.find('Link').length).toBe(1);
  }) 
});