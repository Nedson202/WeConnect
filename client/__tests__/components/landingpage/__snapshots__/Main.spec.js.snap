// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should render 1`] = `
<div>
  <div
    className="text-center"
    id="main-content"
  >
    <h1>
      Welcome to WeConnect
    </h1>
    <h3>
      Here, we make it easier for you to reach businesses faster
    </h3>
    <Link
      replace={false}
      to="/businesses"
    >
      <button
        className="btn"
        id="explore-button"
      >
        Explore
      </button>
    </Link>
  </div>
</div>
`;
